packages/hash/external-services/kratos/kratos.dev.yml

```diff
@@ -37,6 +37,19 @@ selfservice:
       after:
         password:
           hooks:
+            # TODO - figure out whether this blocks writing to the datastore
```

Comment:
In the [kratos documentation](https://www.ory.sh/docs/kratos/self-service/hooks) a "before persisting" and "after persisting" variations of the "after registration" hook are mentioned.

Ideally we'd use the "before persisting" hook, so we can ensure the kratos identity does not get created if anything goes wrong in the user entity creation webhook call. However, I cannot find how to specify whether this hook is "before persisting" or "after persisting". Have added a TODO here to capture that further exploration is required to determine the exact behaviour of this hook. I did not think it was worth the investment at this stage.

Reply:
Good idea to have a TODO here, but I feel like the TODO doesn't capture what you wrote here. Does the "before persisting" block writing to the datastore?

Reply:
I've update this comment to better reflect our understanding of this hook in https://github.com/hashintel/hash/pull/934/commits/9c1f05d4520839d68b58e23d00e8faa20d3c4ccd